### PR TITLE
24 crud for the users in the admin section

### PR DESCRIPTION
**Pull request by Gavalencia12**
# <=== 24 crud for the users in the admin section ====>
## Pull Request Title
#24  Implement CRUD Functionality for User Management in Admin Section

## Detailed Description
- What changes are being made?
This PR implements the "Read" functionality for managing users in the admin section. Admins can now view a list of users, with sorting and filtering options based on attributes like role or status. This is the first step toward full CRUD operations for user management.
- Why are these changes necessary?
These changes are required to allow admins to efficiently manage the user base by viewing essential user details. This is crucial for account oversight, status updates, and role-based management.
- How have these changes been tested?
Test cases have been written to ensure the correct behavior of the new user list functionality. Unit tests verify data synchronization between the local database and Firebase, as well as access control checks to ensure only admins can access this section.
## List of Changes
- User List View:
    A user list view has been added, allowing admins to see all users.
    Sorting and filtering options based on attributes like user role and account status.

- ViewModel:
    Integrated UserViewModel for user management operations in the admin section.
    Linked user data from both the local database and Firebase using FirebaseAuthDataSource.

- Data Mapping:
    Utilized existing UserEntity and UserRole models for consistent data mapping.

- Security Measures:
    Added access control to prevent unauthorized users from accessing the admin user management section.
Testing:
    Unit tests created for the "Read" functionality to ensure synchronization and correct data presentation.
## Additional Information
image
image
image
image

### CODE REVIEW

File: app/src/main/java/com/example/carhive/Presentation/admin/view/Adapters/UserAdapter.kt
Comment (IsmaelMontiel710) [inline app/src/main/java/com/example/carhive/Presentation/admin/view/Adapters/UserAdapter.kt:1]: the role of the user in english please, and comments in english

File: app/src/main/java/com/example/carhive/Presentation/admin/view/Adapters/UserAdapterBan.kt
Comment (IsmaelMontiel710) [inline app/src/main/java/com/example/carhive/Presentation/admin/view/Adapters/UserAdapterBan.kt:1]: also here and the verificartion

File: app/src/main/java/com/example/carhive/Presentation/admin/view/AdminHomeFragment.kt
Comment (IsmaelMontiel710) [inline app/src/main/java/com/example/carhive/Presentation/admin/view/AdminHomeFragment.kt:1]: remove data from being printed in the terminal

File: app/src/main/java/com/example/carhive/Presentation/admin/view/AdminUserActiveFragment.kt
Comment (IsmaelMontiel710) [inline app/src/main/java/com/example/carhive/Presentation/admin/view/AdminUserActiveFragment.kt:1]: Also here and the comments in english

File: app/src/main/java/com/example/carhive/Presentation/admin/view/AdminUserBanFragment.kt
Comment (IsmaelMontiel710) [inline app/src/main/java/com/example/carhive/Presentation/admin/view/AdminUserBanFragment.kt:1]: also here, remove data from being printed in the terminal and comments in english

File: app/src/main/java/com/example/carhive/Presentation/admin/view/AdminUserListFragment.kt
Comment (IsmaelMontiel710) [inline app/src/main/java/com/example/carhive/Presentation/admin/view/AdminUserListFragment.kt:1]: also here, remove data from being printed in the terminal and comments in english

**Review by IsmaelMontiel710 — CHANGES_REQUESTED** [review]
Comments in Spanish, data printed on console and words in Spanish, please correct them

**Review by Angel591Diaz — APPROVED** [review]
You have many errors in your pull request, but as you are small, we will let you pass so you can make another pull request catalogued as bug with the corrections made by our teammate Ismael.

**Review by Kamila04 — APPROVED** [review]

**Pull request merged**